test/client/fixtures/sandbox/event/unload-test.js

```diff
@@ -0,0 +1,29 @@
+asyncTest('BEFORE_UNLOAD_EVENT must be called last (GH-400)', function () {
+    var iframe = document.createElement('iframe');
+
+    iframe.src = window.QUnitGlobals.getResourceUrl('../../../data/unload/iframe.html');
+
+    window.QUnitGlobals.waitForIframe(iframe)
+        .then(function () {
+            var window = iframe.contentWindow;
+            var result = '';
+
+            window.unloadSandbox.on(window.unloadSandbox.BEFORE_UNLOAD_EVENT, function onBeforeUnload () {
+                strictEqual(result, '12');
+
+                start();
+            });
+
+            window.addEventListener(window.unloadSandbox.beforeUnloadEventName, function () {
+                result += '1';
+            });
+
+            window.addEventListener(window.unloadSandbox.beforeUnloadEventName, function () {
```

Comment:
do we really need to add two handlers via `window.addEventListener`? It seems nothing is changed if we leave just one.
And should we check the handler added by `window.onbeforeunload|window.onpagehide` instead the second `addEventListener`?